### PR TITLE
#175743327 ft(manager): Avail requests to manager for approval

### PR DESCRIPTION
**Pull request by SamuelSabasNayo**
#### What does this PR do?
- Avail requests to a manager for approval or rejection.

#### Description of Task to be completed? 
- Allows a manager to get, add, and update requests.
- Implement API documentation by using Swagger.
- Test all APIs by using mocha/chai.

#### How should this be manually tested?
- Clone the repository.

- In your home directory move into the cloned folder by cd this branch **`ft-manager-#175743327`**.

- Then run **`npm install`** to install all dependencies.

- Run **`npm migrate`** to create tables.

- Run **`npm seed`** to insert some dummy data.

- Run **`npm start`** to start the server.

- In your default browser navigate to **`http://http://localhost:5000/api-docs`** to assess the endpoints regarding to manager.

- Run **`npm test`** to test the codes.

#### Any background context you want to provide?
- To successfully get, add, and update requests as a manager, do the following:
	- Signup as a new user **`manager@bn.com`**.
	- Sign in as a Super Admin **`spacenomad@gmail.com`**.
	- Assign a role to the registered user (`manager@bn.com`) to become **`MANAGER`**.

	- Sign in as manager **`manager@bn.com`**.
- As a manager, you can get, add, and update requests.

#### What are the relevant Pivotal Tracker stories?
- [#175743327](https://www.pivotaltracker.com/story/show/175743327)

#### Screenshots (if appropriate)
- 
Screen Shot 2021-01-24 at 14 52 23

#### Questions:
- N/A

### CODE REVIEW

File: src/controllers/managerController.js [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+  // create a request
+  static async addRequest(req, res) {
+    try {
+      const savedRequest = await request.create(req.body);
+
+      return res.status(201).json({ message: 'Request added successfully!', savedRequest });
+    } catch (error) {
+      return res.status(500).json({ error: error.message });
+    }
+  }
```
Comment (SimonImanigirimpuhwe) [inline src/controllers/managerController.js]: I can see here you can create one request a million times with unique data....  and add `idUser` manually!

File: src/tests/manager.test.js [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+                tokenManager = res.body.token;
+              });
+
+              describe('GET /manager/requests', () => {
+                it('Manager should get all requests', async () => {
+                  const res = await request(app)
+                    .get('/manager/requests')
+                    .set('authorization', tokenManager);
+
+                  console.log(res.body);
```
Comment (SimonImanigirimpuhwe) [inline src/tests/manager.test.js]: remove this console.log

**Review by SimonImanigirimpuhwe — CHANGES_REQUESTED** [review]

**Review by ConstantineMaliza — APPROVED** [review]
Great work @SamuelSabasNayo work those change @SimonImanigirimpuhwe  said other tests work well

File: src/controllers/managerController.js (hunk 1 above)
Comment (SamuelSabasNayo) [reply src/controllers/managerController.js]: It is dedicated to someone else.

**Review by kabano89 — APPROVED** [review]

**Review by SimonImanigirimpuhwe — APPROVED** [review]
Good implementation

**Pull request merged**